src/components/GardensDashboard.js

```diff
@@ -1,27 +1,122 @@
-import React from 'react'
-import { GU, Link } from '@1hive/1hive-ui'
+import React, { useCallback } from 'react'
+import { GU, shortenAddress, textStyle, useTheme } from '@1hive/1hive-ui'
 import { useGardens } from '../providers/Gardens'
+import { useHistory } from 'react-router'
+
+import defaultGardenSvg from '../assets/defaultGarden.svg'
+import defaultTokenSvg from '../assets/defaultToken.svg'
 
 function GardensDashboard() {
   const { gardens } = useGardens()
+
   return (
     <div
       css={`
         padding: ${3 * GU}px;
         display: grid;
-        grid-template-columns: repeat(3, 1fr);
+        grid-gap: ${2 * GU}px;
+        grid-template-columns: repeat(4, 1fr);
       `}
     >
       {gardens.map((garden, index) => (
-        <div key={index}>
-          <Link href={`#/garden/${garden.address}`} external={false}>
-            {garden.id}
-          </Link>
-          <span>{garden.proposalCount}</span>
-        </div>
+        <GardenCard key={garden.id} garden={garden} />
       ))}
     </div>
   )
 }
 
+function GardenCard({ garden }) {
+  const theme = useTheme()
+  const history = useHistory()
+  const handleSelectGarden = useCallback(() => {
+    history.push(`/garden/${garden.address}`)
+  }, [garden, history])
+
+  return (
+    <div
+      onClick={handleSelectGarden}
+      css={`
+        padding: ${5 * GU}px ${4 * GU}px ${3 * GU}px ${4 * GU}px;
+        background: ${theme.surface};
+        border: 1px solid ${theme.border};
+        border-radius: ${2.5 * GU}px;
+        box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.15);
+        cursor: pointer;
+
+        display: grid;
+        grid-template-rows: 72px 32px 72px auto auto;
+        grid-gap: ${2 * GU}px;
+        text-align: center;
+      `}
+    >
+      <div
+        css={`
+          height: 72px;
+          display: flex;
+          align-items: center;
+          justify-content: center;
+        `}
+      >
+        <img src={garden.logo || defaultGardenSvg} alt="" />
```

Comment:
lets create a memo with ` const cardData = useMemo(() => {logo: garden.logo || defaultGardenSvg, name: garden.name || shortenAddress(garden.id) ........}, [gardens])`

Reply:
What benefit do you see in this case? I think having a `useMemo` would result in more expensive computation than the simple operations we are performing.

Reply:
well discard the use memo, and we can just have the variable instanced before the html so we don't have the `||` operator and we use the variable directly